pkg/boot/diskboot/parser.go

```diff
@@ -115,9 +125,9 @@ func (p *parser) parseGrubEntry(line string) {
 		}
 		p.entry.Modules = append(p.entry.Modules,
 			NewModule(filteredParams[1], filteredParams[2:]))
-	case "linux":
+	case matchPrefix("linux.*", prefix):
```

Comment:
is this what grub does? or does it to "linux", "linux16", "linux32": ?

Reply:
depending on grub versions (or enabled build feature) it has: "linux", "linux16", "linuxefi" (no linux32)
https://github.com/JulienVdG/u-root/blob/31e92bed662301419a14ba665fcc529fb83398d9/pkg/boot/grub/grub.go#L337

Reply:
actually do we want to make this change or is it better to wait until #1469 is merged in and then use the parser from there?

Reply:
I'd say anything that adds examples/tests and exposes features that need to be covered to boot real-world distros is always worth adding.

Reply:
Exactly, I am trying to compare the different implementations and add missing bits into the new packages with the end goal to refactor them all.
I started by writing a grub parser that followed the syslinux/pxelinux parser structure, but there are still some missing bits... but waiting to long before review leads to difficult and long review, so please add tests now so we don't forget about this case.